estela-api/api/utils.py

```diff
@@ -29,9 +29,10 @@ def update_env_vars(instance, env_vars, level="project"):
             elif level == "spider":
                 SpiderJobEnvVar.objects.create(spider=instance, **env_var)
 
-    for env_var in env_vars_instance:
-        if env_var.name not in [value["name"] for value in env_vars]:
-            env_var.delete()
+    if delete:
+        for env_var in env_vars_instance:
+            if env_var.name not in [value["name"] for value in env_vars]:
+                env_var.delete()
```

Comment:
Why add a `delete` argument for this function?

Reply:
It's used here: https://github.com/bitmakerla/estela/blob/add-bmc-proxies/estela-api/api/views/proxyprovider.py#L67